test/LocalNeuralNetTest.cpp

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2015 - 2022, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+#include "geopm/Exception.hpp"
+
+#include "LocalNeuralNet.hpp"
+
+class LocalNeuralNetTest : public ::testing::Test
+{
+    protected:
+        void SetUp();
+        LocalNeuralNet net;
+};
+
+void LocalNeuralNetTest::SetUp()
+{
+    net = LocalNeuralNet (
+            json11::Json::array {
+                json11::Json::array {
+                    json11::Json::array {  json11::Json::array {8, 16} },
+                    json11::Json::array {3}
+                },
+                json11::Json::array {
+                    json11::Json::array { json11::Json::array { 1 } },
+                    json11::Json::array {0}
+                }
+            }
+          );
+}
+
+TEST_F(LocalNeuralNetTest, test_inference) {
+    TensorOneD inp(2);
+    inp[0] = 1;
+    inp[1] = 2;
+    TensorOneD out = net.model(inp);
+    EXPECT_EQ(1, out.get_dim());
```

Comment:
I am confused about the use of `EXPECT_EQ` for *floating point values*.
This is because floating point values cannot be compared exactly, and instead we take the difference of the two values, and see if it is within the allowable range of *epsilon* difference. In other words, are the two values *close enough*?

For example,
```cpp
float x = 1.0;
float y = 1.00001;
bool b = x == y;  // this is false
```

I do not  know how the `EXPECT_EQ` macro is implemented. But the main point is that we cannot perform integer comparisons on floating point values. Please see if there is a floating point equivelent macro in the google test library.

Or you can use this:
```cpp
#define EPSILON 0.0001  // what is "close enough" is dictated by requirements
EXPECT_EQ( (fabs(1 - out.get_dim()) < EPSILON), true );

// this is the general form
// if (fabs(number1 - number2) < EPSILON) {}
```

Reply:
You can use the `std::numeric_limits<float>::epsilon()` also.

https://en.cppreference.com/w/cpp/types/numeric_limits/epsilon

Reply:
Nice catch! We'll replace with EXPECT_NEAR using an epsilon of 1e-6.